src/peft/peft_model.py

```diff
@@ -350,7 +356,34 @@ def from_pretrained(
             model = cls(model, config, adapter_name)
         else:
             model = MODEL_TYPE_TO_PEFT_MODEL_MAPPING[config.task_type](model, config, adapter_name)
-        model.load_adapter(model_id, adapter_name, is_trainable=is_trainable, **kwargs)
+
+        if isinstance(model.base_model, xLoRAModel):
+            assert isinstance(config, xLoRAConfig)
```

Comment:
In general, let's not use `assert`s, always raise a proper error with a nice error message.

Reply:
I have added a nice exception raise when applicable, thank you!